dashboard/app/models/ability.rb

```diff
@@ -81,6 +81,11 @@ def initialize(user)
       can :index, Section, user_id: user.id
       can [:get_feedbacks, :count, :increment_visit_count, :index], TeacherFeedback, student_id: user.id
 
+      can :list_projects, Section do |section|
+        return true if can? :manage, section
+        user.sections_as_student.include?(section)
```

Comment:
for extra credit, I think this is simple enough to put on one line. the body of this block can just be:
```
can?(:manage, section) || user.sections_as_student.include?(section)
```

Reply:
👍